File: Projects/Framework-Samples/Csla/CSharp/PetShop.Silverlight/Source/PetShop.UI.Silverlight/ViewModel/CategoryViewModel.cs
using System;
using System.Net;
using System.Windows;
using System.Windows.Controls;
using System.Windows.Documents;
using System.Windows.Ink;
using System.Windows.Input;
using System.Windows.Media;
using System.Windows.Media.Animation;
using System.Windows.Shapes;

using Csla.Xaml;

using PetShop.Business;

namespace PetShop.UI.Silverlight
{

    public class CategoryViewModel : ViewModel<Category>
    {
        /// <summary>
        /// Gets or sets the Test object.
        /// </summary>
        public static readonly DependencyProperty TestProperty = DependencyProperty.Register("Test", typeof(object), typeof(CategoryViewModel),
            new PropertyMetadata((o, e) =>
            {
                ((CategoryViewModel)o).Model = (Category)e.NewValue;
            }));
        
        /// <summary>
        /// Gets or sets the Test object.
        /// </summary>
        public object Test
        {
            get { return GetValue(TestProperty); }
            set { SetValue(TestProperty, value); }
        }
    }
}

File: tags/CSLA-4.0-Samples/VisualBasic/PetShop/Source/PetShop.UI.Silverlight/ViewModels/CategoryViewModel.vb
Imports System.Net
Imports System.Windows
Imports System.Windows.Controls
Imports System.Windows.Documents
Imports System.Windows.Ink
Imports System.Windows.Input
Imports System.Windows.Media
Imports System.Windows.Media.Animation
Imports System.Windows.Shapes

Imports Csla.Xaml

Imports PetShop.Business

Public Class CategoryViewModel
    Inherits ViewModel(Of Category)
    ''' <summary>
    ''' Gets or sets the Test object.
    ''' </summary>
    Public Shared ReadOnly TestProperty As DependencyProperty = DependencyProperty.Register("Test", GetType(Object), GetType(CategoryViewModel), New PropertyMetadata(Sub(o, e)
                                                                                                                                                                          DirectCast(o, CategoryViewModel).Model = DirectCast(e.NewValue, Category)
                                                                                                                                                                      End Sub))

    ''' <summary>
    ''' Gets or sets the Test object.
    ''' </summary>
    Public Property Test() As Object
        Get
            Return GetValue(TestProperty)
        End Get
        Set(ByVal value As Object)
            SetValue(TestProperty, value)
        End Set
    End Property
End Class